roles/contrail_master/tasks/os_contrail_master.yaml

```
@@ -0,0 +1,108 @@
+---
+- name: Create the get IP file to convert hostname to ip
+  template:
+    src: getIp.j2
+    dest: /tmp/getIp.py
+    owner: root
+    mode: 0644
+  run_once: True
+
+- set_fact:
+    api_vip: "{{ groups.lb.0 | ipaddr }}"
+  when: groups.lb is defined
+  run_once: True
+
+- set_fact:
+    api_vip: "{{ openshift_master_cluster_public_hostname }}"
+  when: openshift_master_cluster_public_hostname is defined
+  run_once: True
+
+- set_fact:
+    api_vip: "{{ groups.masters.0 }}"
+  when: api_vip is not defined
+  run_once: True
+
+- name: validate api_vip hostname or ip
+  command: python /tmp/getIp.py {{ api_vip }}
+  register: api_vip_result
+  run_once: True
+
+- name: set api_vip
+  set_fact:
+    api_vip: "{{ api_vip_result.stdout }}"
+  when: api_vip_result is success
+  run_once: True
+
+- name: Create the contrail namespace template file
+  template:
+    src: contrail-namespace.j2
+    dest: /tmp/contrail-namespace.yaml
+    owner: root
+    mode: 0644
+  run_once: True
+
+- name: Create the contrail-system namespace
+  command: "{{ openshift_client_binary }} create -f /tmp/contrail-namespace.yaml"
```

```suggestion
  oc_project:
    state: present
    name: "contrail-system"
```

This would make `roles/contrail_master/templates/contrail-namespace.j2` obsolete and avoid unsafe `ignore_errors: True`